qiskit/ignis/verification/randomized_benchmarking/dihedral.py

```diff
@@ -429,19 +434,19 @@ def cnot(self, i, j):
         """
         assert i >= 0, "i negative!"
         assert j >= 0, "j negative!"
-        assert i < self.n_qubits, "i too big!"
-        assert j < self.n_qubits, "j too big!"
+        assert i < self.num_qubits, "i too big!"
+        assert j < self.num_qubits, "j too big!"
         assert i != j, "i == j!"
         self.linear[j] = [(self.linear[i][k] + self.linear[j][k]) % 2
```

Comment:
I won't leave comments on all of them, but in general rewriting all these for loop methods to use numpy arrayfunctions will give large performance improvements.